core/model/src/main/java/org/eclipse/rdf4j/model/util/Values.java

```diff
@@ -0,0 +1,694 @@
+	 * 
+	 * @throws NullPointerException if statement is <code>null</code>.
+	 */
+	public static Triple triple(Statement statement) {
+		Objects.requireNonNull(statement, "statement may not be null");
+		return VALUE_FACTORY.createTriple(statement.getSubject(), statement.getPredicate(), statement.getObject());
+	}
+
+	/**
+	 * Creates a new {@link Triple RDF* embedded triple} using the subject, predicate and object from the supplied
+	 * {@link Statement}.
+	 * 
+	 * @param vf        the {@link ValueFactory} to use for creation of the {@link Triple}
+	 * @param statement the {@link Statement} from which to construct a {@link Triple}
+	 * 
+	 * @return a {@link Triple} with the same subject, predicate, and object as the supplied Statement.
+	 * 
+	 * @throws NullPointerException if any of the supplied input parameters is <code>null</code>.
+	 */
+	public static Triple triple(ValueFactory vf, Statement statement) {
+		Objects.requireNonNull(statement, "statement may not be null");
+		return vf.createTriple(statement.getSubject(), statement.getPredicate(), statement.getObject());
+	}
+
+	/**
+	 * Get a {@link ValueFactory}.
+	 * 
+	 * @return a {@link ValueFactory}.
+	 */
+	public static ValueFactory getValueFactory() {
+		return VALUE_FACTORY;
+	}
+
+	/* private methods */
+
+	/**
+	 * Creates a typed {@link Literal} out of the supplied object, mapping the runtime type of the object to the
+	 * appropriate XML Schema type. If no mapping is available, the method throws an exception if the boolean parameter
+	 * is true, or if it is false it returns a literal with the string representation of the supplied object as the
+	 * value, and {@link XSD#STRING} as the datatype. Recognized types are {@link Boolean}, {@link Byte},
+	 * {@link Double}, {@link Float}, {@link Integer}, {@link Long}, {@link Short}, {@link XMLGregorianCalendar } , and
+	 * {@link Date}.
+	 *
+	 * @param valueFactory            The {@link ValueFactory} to use when creating the result.
+	 * @param object                  an object to be converted to a typed literal.
+	 * @param throwExceptionOnFailure If true throws a {@link LiteralUtilException} when the object is not recognised.
+	 *                                If false it returns a string typed literal based on the objects toString method.
+	 * @return a typed literal representation of the supplied object.
+	 * @throws IllegalArgumentException If the literal could not be created.
+	 * @throws NullPointerException     If the object was null.
+	 */
+	private static Literal createLiteralFromObject(ValueFactory valueFactory, Object object,
+			boolean throwExceptionOnFailure)
+			throws IllegalArgumentException {
+		Objects.requireNonNull(valueFactory, "valueFactory may not be null");
+		Objects.requireNonNull(object, "object may not be null");
+
+		if (object instanceof Boolean) {
+			return valueFactory.createLiteral(((Boolean) object).booleanValue());
+		} else if (object instanceof Byte) {
```

Comment:
I am not a 100% sure but it may a good idea to also test for the primitive types here, i.e. `byte.class.equals(object.getClass()) `, similar for int, double etc

Might also be a good idea to cover this in tests if not already done so

Reply:
The argument here can never be a primitive type, since it is an `Object`. Instead, to cover primitives, the `literal` method has been overloaded with functions that _explicitly_ expect a primitive type. 

I think most of this is covered in the tests as well.

It's perhaps worth pointing out that due to autoboxing, we wouldn't even need the `literal(Object)` method if all we wanted was to support feeding in an object that corresponds to a primitive type: having the methods accept primitives is enough for that, the compiler would take care of the rest for us. 

What this `literal(Object)` method gives us extra is that you can feed it _any_ object, without needing to do type checking first, and you get back a literal.